bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/CommerceLinksTransformerFactory.java

```diff
@@ -115,25 +139,147 @@ public void startElement(String uri, String localName, String qName, Attributes
                 return;
             }
 
+            boolean replaceText = Boolean.parseBoolean(attributes.getValue(ATTR_REPLACE_TEXT));
+            String newHref = null;
             String productSku = attributes.getValue(ATTR_PRODUCT_SKU);
-            AttributesImpl attributesImpl = new AttributesImpl(attributes);
             if (StringUtils.isNotBlank(productSku)) {
                 // if there is both product and category attribute on a link then product attribute is honored
                 Page currentPage = request.getResourceResolver().adaptTo(PageManager.class).getContainingPage(request.getResource());
                 Page productPage = SiteNavigation.getProductPage(currentPage);
-                String newHref = urlProvider.toProductUrl(request, productPage, productSku);
-                attributesImpl.setValue(attributes.getIndex(ATTR_HREF), newHref);
+                if (replaceText) {
+                    newHref = prepareProductInfo(productSku, productPage);
+                } else {
+                    newHref = urlProvider.toProductUrl(request, productPage, productSku);
+                }
             } else {
                 String categoryUid = attributes.getValue(ATTR_CATEGORY_UID);
                 if (StringUtils.isNotBlank(categoryUid)) {
                     Page currentPage = request.getResourceResolver().adaptTo(PageManager.class).getContainingPage(request.getResource());
                     Page categoryPage = SiteNavigation.getCategoryPage(currentPage);
-                    String newHref = urlProvider.toCategoryUrl(request, categoryPage, categoryUid);
-                    attributesImpl.setValue(attributes.getIndex(ATTR_HREF), newHref);
+                    if (replaceText) {
+                        newHref = prepareCategoryInfo(categoryUid, categoryPage);
+                    } else {
+                        newHref = urlProvider.toCategoryUrl(request, categoryPage, categoryUid);
+                    }
+                }
+            }
+
+            if (StringUtils.isNotBlank(newHref)) {
+                AttributesImpl newAttributes = new AttributesImpl(attributes);
+                newAttributes.setValue(attributes.getIndex(ATTR_HREF), newHref);
+                if (StringUtils.isNotBlank(linkText)) {
+                    String title = attributes.getValue(ATTR_TITLE);
+                    if (title == null) {
+                        // set title to linkText
+                        newAttributes.addAttribute("", ATTR_TITLE, ATTR_TITLE, "CDATA", linkText);
+                    }
+                }
+                super.startElement(uri, localName, qName, newAttributes);
+                if (StringUtils.isNotBlank(linkText)) {
+                    char[] chars = linkText.toCharArray();
+                    // insert linkText as new content
+                    super.characters(chars, 0, chars.length);
```

Comment:
You could also write the end endElement here directly. So you would not have to keep track on the nesting level of the <a> tags descendants but could simply ignore all up to and including the original endElement tag.

Reply:
I don't see how that would work. The endElement() will be called anyway when the parsing reaches </a> of the commerce link and then we need to stop ignoring content by tracking the nesting level.